packages/core/lib/utils.js

```diff
@@ -168,6 +168,10 @@ var utils = {
       if (!traceData) {
         traceData = {};
         logger.getLogger().error('_X_AMZN_TRACE_ID is empty or has an invalid format');
+      } else if (traceData.root && !traceData.parent && !traceData.sampled) {
+        // Lambda PassThrough only has root, treat as valid in this case and mark the segment
+        segment.noOp = true;
+        valid = true;
```

Comment:
If in `"valid no-op"` scenario, do we want to ensure that the segment/subsegment is not being flushed to XRay Daemon/FluxPump, and if so, is this ensured in this PR?

Reply:
https://github.com/aws/aws-xray-sdk-node/blob/dd12415098a5cb4d9846683cae9508e857fa8835/packages/core/lib/utils.js#L171-L190
- If `no-op`, then `notTraced` is true, which means that segment is not sent